### PR TITLE
Fix #9224 Multiple issues with jobs when logged in via OIDC with SU role

### LINKED ISSUE #9224: Multiple issues with jobs when logged in via OIDC with SU role
### How to Reproduce
1. Log in with OIDC user with SU role attached
2. Either:
  a. Try to remove a root package in the Navigator
  b. Try to download an entity type in the Data Explorer 2

### Expected behavior
Actions succeed.

### Observed behavior
Client keeps polling indefinitely. Logs show that the jobs can't be updated because of insufficient permissions, e.g.:
```
WARN  o.m.jobs.JobExecutionUpdaterImpl - Error updating job execution
org.molgenis.data.security.exception.EntityTypePermissionDeniedException: permission:UPDATE_DATA entityTypeId:sys_job_ResourceDownloadJobExecution
```

### PR DESCRIPTION
**Pull request by tommydeboer**
#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] Migration step added in case of breaking change
- [x] User documentation updated
- [x] (If you have changed REST API interface) view-swagger.ftl updated
- [x] Test plan template updated
- [x] Clean commits
- [x] Added Feature/Fix to release notes

### CODE REVIEW

**Comment by tommydeboer** [comment]
[Fix #9224 Job issues with SU OIDC login](https://trello.com/c/1BU2KOaB/93-fix-9224-job-issues-with-su-oidc-login)

File: molgenis-jobs/src/main/java/org/molgenis/jobs/JobExecutor.java [hunk 1]
```diff
@@ -178,16 +185,30 @@ private Job saveExecutionAndCreateJob(JobExecution jobExecution) {
...
+          createJobExecutionContext(jobExecution, authentication);
+
       jobExecutionTemplate.call(job, progress, jobExecutionContext);
     } finally {
       JobUtils.cleanupAfterRunJob(callingThreadId);
     }
   }
 
+  private JobExecutionContext createJobExecutionContext(
+      JobExecution jobExecution, @Nullable Authentication authentication) {
```
Comment (fdlk) [inline molgenis-jobs/src/main/java/org/molgenis/jobs/JobExecutor.java:204]: Consider making it an Optional

File: molgenis-jobs/src/main/java/org/molgenis/jobs/JobExecutionContextFactory.java [hunk 2]
```diff
@@ -1,8 +1,12 @@
 package org.molgenis.jobs;
 
 import org.molgenis.jobs.model.JobExecution;
+import org.springframework.security.core.Authentication;
 
 public interface JobExecutionContextFactory {
 
   JobExecutionContext createJobExecutionContext(JobExecution jobExecution);
+
+  JobExecutionContext createJobExecutionContext(
```
Comment (fdlk) [inline molgenis-jobs/src/main/java/org/molgenis/jobs/JobExecutionContextFactory.java:10]: Let's give this one a different name, like createJobExecutionContextWithAuthentication or so

File: molgenis-jobs/src/main/java/org/molgenis/jobs/JobExecutionContextFactoryImpl.java [hunk 3]
```diff
@@ -20,9 +20,28 @@
     this.jobExecutorLocaleService = requireNonNull(jobExecutorLocaleService);
   }
 
+  /**
+   * Creates a JobExecutionContext with an {@link Authentication} based on the {@link JobExecution}.
+   *
+   * @param jobExecution the JobExecution to create a context for
+   * @return a JobExecutionContext
+   */
   @Override
```
Comment (fdlk) [inline molgenis-jobs/src/main/java/org/molgenis/jobs/JobExecutionContextFactoryImpl.java:29]: Move the javadoc to the interface

**Review by fdlk — CHANGES_REQUESTED** [review]
The commit description is smaller than the changes, I think.

**Pull request merged**